src/CommunicationExecutor.c

```diff
@@ -207,6 +207,62 @@ void Ros_Communication_PublishActionFeedback(rcl_timer_t* timer, int64_t last_ca
     Ros_ActionServer_FJT_ProcessResult();
 }
 
+void Ros_Communication_MonitorUserLanState(SHORT interface_id)
+{
+    //assume link is currently up (and pretend it was up before this task even
+    //started).
+    //This task is only started once MotoROS2 can reach the Agent (and can
+    //connect to it), so the link MUST be up or otherwise this task could
+    //not have been started.
+    //If we start with the assumption the link is down, we'd see a state change
+    //as soon as we call Ros_UserLan_IsLinkUp(..), which would not lead to any
+    //action (as we only act on up->down edges), but would be unnecessary.
+    BOOL bLinkWasUp = TRUE;
+
+    //we set g_Ros_Communication_AgentIsConnected=FALSE in the body below,
+    //so this exits when a link drop is detected
+    while (g_Ros_Communication_AgentIsConnected)
+    {
+        //TODO(gavanderhoorn): make configurable ?
+        Ros_Sleep(0.5 * 1e3);
+
+        //retrieve current state of monitored link, assume it's down
+        BOOL bLinkIsUp = FALSE;
+        STATUS status = Ros_UserLan_IsLinkUp(interface_id, &bLinkIsUp);
+        if (status != OK)
+        {
+            //TODO(gavanderhoorn): should this be fatal for MotoROS2?
+            Ros_Debug_BroadcastMsg(
+                "Error reading link state (port: %d, error: %d), shutting down monitor",
+                interface_id, status);
+            //also to console, in case link is actually down
+            Ros_Debug_LogToConsole(
+                "Error reading link state (port: %d, error: %d), shutting down monitor",
+                interface_id, status);
+            //yes, goto has its uses
+            goto UserLanMonitorExit;
+        }
+
+        //link was up last time, but no longer is up now -> signal disconnect
+        if (bLinkWasUp && !bLinkIsUp)
+        {
+            //log to console, as network connection is presumable down
+            Ros_Debug_LogToConsole(
+                "Link down (port: %d), flagging Agent as disconnected", interface_id);
```

Comment:
We can't UDP broadcast here, as the link is down.

I also didn't want to have to set `log_to_stdout: true` for someone to be able to see this.

This is now one of a handful of log msgs which make it to `stdout`.